app/src/main/java/com/sdp/movemeet/utility/ImageHandler.java

```diff
@@ -0,0 +1,71 @@
+package com.sdp.movemeet.utility;
+
+import android.net.Uri;
+import android.util.Log;
+import android.view.View;
+import android.widget.ProgressBar;
+
+import androidx.annotation.NonNull;
+
+import com.google.android.gms.tasks.OnFailureListener;
+import com.google.android.gms.tasks.OnSuccessListener;
+import com.google.android.gms.tasks.Task;
+import com.google.firebase.storage.UploadTask;
+import com.sdp.movemeet.backend.BackendManager;
+import com.sdp.movemeet.backend.firebase.storage.StorageImageManager;
+import com.sdp.movemeet.models.Image;
+import com.squareup.picasso.Picasso;
+
+public class ImageHandler {
+
+    private static final String TAG = "FirebaseInteraction";
+
+    private static BackendManager<Image> imageBackendManager;
+
+    public static void loadImage(Image image, ProgressBar progressBar) {
```

Comment:
Might have to think about the effectiveness of the Image class, and the fact that only having the direct call to ImageView, ImagePath and ImageUri might be enough.

Reply:
Yes, you're right... As we discussed during our last standup meeting, it might not be worth keeping this `Image` class in the future if we only make direct calls to imageImageView, imagePath and imageUri. Let me know how your cache implementation is evolving and, if we have time and if it makes sense, we can simply remove this `Image` class!